shell/common/shell.cc

```diff
@@ -1240,4 +1251,17 @@ Rasterizer::Screenshot Shell::Screenshot(
   return screenshot;
 }
 
+bool Shell::WaitForFirstFrame(fml::TimeDelta timeout) {
+  FML_DCHECK(is_setup_);
+  FML_DCHECK(!task_runners_.GetUITaskRunner()->RunsTasksOnCurrentThread());
+  FML_DCHECK(!task_runners_.GetGPUTaskRunner()->RunsTasksOnCurrentThread());
+  std::unique_lock<std::mutex> lock(waiting_for_first_frame_mutex_);
+  bool success = waiting_for_first_frame_condition_.wait_for(
+      lock, std::chrono::milliseconds(timeout.ToMilliseconds()),
+      [& waiting_for_first_frame = waiting_for_first_frame_] {
+        return !waiting_for_first_frame.load();
+      });
+  return !success;
```

Comment:
Thanks to the tests added, I finally realized that this returns `!success` instead of `success`.

This is a little counter-intuitive as most `wait_for_something` returns `success`. Maybe create an `enum FirstFrameWaitingResult {timedout, successful};` and make that as the return type. That will make it much clearer.

Reply:
I tend to agree, but I was being consistent with the rest of our API: https://github.com/flutter/engine/blob/7f4f52f95294544a2970e514d163b9791efa9147/fml/synchronization/waitable_event.h#L58

Reply:
Good finding. I think we probably should change that return type to enum too to reduce confusions.

Reply:
Ok, i've added the Status class as discussed.